core/src/main/java/org/apache/iceberg/stats/BlobMetadata.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.stats;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Objects;
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+
+public class BlobMetadata {
+  private final String type;
+  private final Set<Integer> columns;
+  private final long offset;
+  private final long length;
+  private final String compressionCodec;
+
+  @JsonCreator
+  public BlobMetadata(
+      @JsonProperty("type") String type,
+      @JsonProperty("columns") Set<Integer> columns,
+      @JsonProperty("offset") long offset,
+      @JsonProperty("length") long length,
+      @JsonProperty("compression_codec") @Nullable String compressionCodec) {
+    this.type = Preconditions.checkNotNull(type, "type is null");
+    this.columns = ImmutableSet.copyOf(Preconditions.checkNotNull(columns, "columns is null"));
+    this.offset = offset;
+    this.length = length;
+    this.compressionCodec = compressionCodec;
+  }
+
+  public String type() {
+    return type;
+  }
+
+  public Set<Integer> columns() {
```

Comment:
This is the set of fieldIds in the Iceberg table this blob applies to? (Probably should doc that) If these are field ID's is the association between this object and the table made at another level? (I'll probably find out if I keep reading)

Reply:
yes and yes